frontend/src/metabase/query_builder/components/DataSelector/saved-entity-picker/SavedEntityList.tsx

```diff
@@ -47,6 +49,10 @@ const SavedEntityList = ({
       : skipToken,
   );
   const list = data?.data ?? [];
+  const filteredList = databaseId
```

Comment:
This is a new fix, this bug has existed since 49, the last version that this data picker was still used for internal use cases.

I was considering 2 other options. Given this component is using `/api/collection/:id/items`.
1. Using the search enpdoint instead, since it supports `table_db_id`, but it doesn't support `collection` (ID)
2. passing database ID somehow to the `/api/collection/:id/items` endpoint, but that doesn't seem possible.

So, I filter the result in FE instead.

Reply:
> So, I filter the result in FE instead.

New data picker does the same :+1: